lightning/src/onion_message/messenger.rs

```diff
@@ -914,7 +914,7 @@ where
 		for (node_id, recipient) in self.message_recipients.lock().unwrap().iter_mut() {
 			if let OnionMessageRecipient::PendingConnection(_, addresses, _) = recipient {
 				if let Some(addresses) = addresses.take() {
```

Comment:
Yeah, it's intentional but not unusable. It is only taking the addresses but leaves the buffered messages so it can be transitioned to `OnionMessageRecipient::ConnectedPeer` once connected. `None` indicates to start ticking the timer. See `OnionMessenger::timer_tick_occurred`.